### PR TITLE
fix: allow whitelisted paths like onboarding as team/user slugs on org domains

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes an issue where certain app routes (like `/onboarding`) couldn't be used as team/user slugs on org domains, resulting in 404 errors on URLs like `acme.cal.com/onboarding`.

image
image

This PR introduces a whitelisting mechanism that allows specific paths to be excluded from the reserved routes list when matching org domain patterns. Currently whitelists the "onboarding" path, allowing teams/users on org domains to use it as their slug while preserving the `/onboarding` app route on non-org domains.

**Link to Devin run:** https://app.devin.ai/sessions/3a92b083cd6c4b2eaaad6116723ce24b

**Requested by:** @hariombalhara (hariom@cal.com)

## Technical Changes

1. **Added `whitelistedPaths` array** in `pagesAndRewritePaths.js` containing paths that should be allowed as slugs on org domains
2. **Updated `getRegExpMatchingAllReservedRoutes`** function to accept an `exclusions` parameter and filter out whitelisted paths
3. **Modified org route patterns** (`orgUserRoutePath`, `orgUserTypeRoutePath`, `orgUserTypeEmbedRoutePath`) to exclude whitelisted paths from their negative lookahead patterns
4. **Added comprehensive tests** to verify the whitelisting behavior works correctly

## Important Review Points

⚠️ **Routing behavior change**: Teams/users on org domains can now use "onboarding" as their slug. If any existing orgs have a team with slug "onboarding", this changes what users see at that URL (they'll see the team page instead of the app's onboarding flow).

🔍 **Regex pattern changes**: The negative lookahead patterns in org routes are complex. Please verify the filtering logic works correctly, especially for edge cases.

📝 **Whitelist scope**: Currently only "onboarding" is whitelisted. Consider whether other paths should also be whitelisted.

## How should this be tested?

1. **Unit tests**: Run `TZ=UTC yarn test apps/web/test/lib/pagesAndRewritePaths.test.ts` and `TZ=UTC yarn test apps/web/test/lib/next-config.test.ts` to verify the tests pass
2. **Manual testing on org domain** (if possible):
   - Create a team with slug "onboarding" on an org domain
   - Visit `<org-domain>.cal.com/onboarding`
   - Verify it routes to the team page (not 404)
3. **Manual testing on non-org domain**:
   - Visit `app.cal.local:3000/onboarding`
   - Verify the app's onboarding flow still works correctly

## Mandatory Tasks

- [x] I have self-reviewed the code
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a documentation change. **N/A** - This is an internal routing fix
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works

## Checklist

- [x] I have read the contributing guide
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked if my changes generate no new warnings

### CODE REVIEW

**Review by alishaz-polymath — COMMENTED** [review]
Approach LGTM, but I defer to Foundation for final decision 🙏

File: apps/web/pagesAndRewritePaths.js [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
+
+/**
+ * Extracts top-level route names from all pages/app files and excludes them from org rewrite.
+ * For example: /abc/def/ghi -> 'abc'
+ * 
+ * These top-level route names are excluded from rewrites in beforeFiles in next.config.js
+ * to prevent conflicts with organization slug rewrites.
+ */
+/* eslint-disable no-undef */
+let topLevelRoutesExcludedFromOrgRewrite = exports.topLevelRoutesExcludedFromOrgRewrite = glob
```
Comment (hariombalhara) [inline apps/web/pagesAndRewritePaths.js:23]: varaible rename from pages -> topLevelRoutesExcludedFromOrgRewrite

File: packages/prisma/schema.prisma [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// This is your Prisma Schema file
+// This is your Prisma Schema file
```
Comment (hariombalhara) [inline packages/prisma/schema.prisma:1]: This was to invalidate cache causing integraiton tests failure

**Review by hbjORbj — APPROVED** [review]
